extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -106,38 +58,18 @@ public KafkaIndexTaskClient(
       long numRetries
   )
   {
-    this.httpClient = httpClient;
-    this.jsonMapper = jsonMapper;
-    this.taskInfoProvider = taskInfoProvider;
-    this.httpTimeout = httpTimeout;
-    this.numRetries = numRetries;
-    this.retryPolicyFactory = createRetryPolicyFactory();
-
-    this.executorService = MoreExecutors.listeningDecorator(
-        Execs.multiThreaded(
-            numThreads,
-            StringUtils.format(
-                "KafkaIndexTaskClient-%s-%%d",
-                dataSource
-            )
-        )
-    );
-  }
-
-  public void close()
-  {
-    executorService.shutdownNow();
+    super(httpClient, jsonMapper, taskInfoProvider, httpTimeout, dataSource, numThreads, numRetries);
   }
 
   public boolean stop(final String id, final boolean publish)
   {
     log.debug("Stop task[%s] publish[%s]", id, publish);
 
     try {
-      final FullResponseHolder response = submitRequest(
+      final FullResponseHolder response = submitRequestWithEmptyContent(
```

Comment:
This is an area of the code I am not at all familiar with, so I don't have much capacity to review changes to the kafka indexing :(

Reply:
In this PR, I just 1) added `IndexTaskClient` class and move some codes from `KafkaIndexClient` to it to use it in both Kafka indexing and parallel batch indexing and 2) added some method like `submitRequestWithEmptyContent`, `submitJsonRequest`, `submitSmileRequest` for convenience. 

BTW, I think no one understands the current implementation of Kafka indexing service completely line by line, so don't worry about it. Hope things are going better after https://github.com/druid-io/druid/issues/5906.